### PR TITLE
move share by mail capabilities to the share provider

### PR DESCRIPTION
**Pull request by schiessle**
make it look the same like on stable11: https://github.com/nextcloud/server/pull/4251

### CODE REVIEW

File: apps/sharebymail/lib/Capabilities.php [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+	 * Function an app uses to return the capabilities
+	 *
+	 * @return array Array containing the apps capabilities
+	 * @since 8.2.0
+	 */
+	public function getCapabilities() {
+		return [
+			'sharebymail' =>
+				[
+					'upload_files_drop' => true,
```
Comment (nickvergessen) [inline apps/sharebymail/lib/Capabilities.php]: `'enabled' => true,` is missing

**Review by nickvergessen — CHANGES_REQUESTED** [review]

**Comment by schiessle** [comment]
updated

File: apps/sharebymail/lib/Capabilities.php [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+	/**
+	 * Function an app uses to return the capabilities
+	 *
+	 * @return array Array containing the apps capabilities
+	 * @since 8.2.0
+	 */
+	public function getCapabilities() {
+		return [
+			'sharebymail' =>
```
Comment (rullzer) [inline apps/sharebymail/lib/Capabilities.php]: this means that it now is outside of the files_sharing block... I don't really care.

ALso we should document it somewhere...

Comment (schiessle) [inline apps/sharebymail/lib/Capabilities.php]: yes, was requested here https://github.com/nextcloud/server/pull/4251

I'm also not sure, we have all other share capabilities in the "files_sharing" block, e.g. also for federated sharing (which would have it's own app as well). But I also don't care that much, that's why I followed @MorrisJobke and  @nickvergessen here.

> Also we should document it somewhere...

I don't think we have a place to document all available capabilities, do we?

Comment (nickvergessen) [inline apps/sharebymail/lib/Capabilities.php]: you can have them in the files_sharing block just fine, as said in the other issue.
As per https://github.com/nextcloud/server/pull/4251#discussion_r110416021
you can nested the array as you want and the array will be merged.

Comment (schiessle) [inline apps/sharebymail/lib/Capabilities.php]: missed that comment somehow... let me try

Comment (schiessle) [inline apps/sharebymail/lib/Capabilities.php]: works indeed 😃 I update both PR's to move it in the "files_sharing" category.

**Review by MorrisJobke — APPROVED** [review]
Tested and works 👍

**Review by rullzer — APPROVED** [review]

**Pull request merged**